src/main/java/com/appdynamics/extensions/filewatcher/FileWatcherMonitorTask.java

```diff
@@ -28,6 +28,8 @@ public void run() {
 		FileWatcherMonitor.getLatch().countDown();
 	}
 
+	//#TODO This method will no longer be needed and have to be referenced from FileWatcherMonitor once you have reporting piece moved in here.
+
 	public static Map<String, String> getFilestoprocess() {
 		return filesToProcess;
```

Comment:
Also, a question arises. Suppose two file paths configured have some common files. Currently when all metrics are being reported together, it is made sure that we dont publish metrics for the same file again and again. If we start reporting metrics in the task, this static map will ensure that common files are not reported multiple times. But will need to use a thread safe map.